test/sessionClaimValidatorRefresh.test.js

```diff
@@ -0,0 +1,129 @@
+            DateProviderReference.init();
+        });
+
+        it("SessionClaimValidator.shouldRefresh should return false after refreshing", function () {
+            const tokenPayload = {
+                "st-test": {
+                    v: false,
+                    t: Date.now() - 20000 // setting claim fetched time to 20 seconds ago
+                }
+            };
+
+            const testClaim = new BooleanClaim({
+                id: "st-test",
+                refresh: () => {
+                    tokenPayload["st-test"].t = Date.now();
+                },
+                defaultMaxAgeInSeconds: 10 /* 10 seconds */
+            });
+
+            const testClaimValidator = testClaim.validators.isTrue();
+
+            assert(testClaimValidator.shouldRefresh(tokenPayload) === true);
+            testClaimValidator.refresh();
+            assert(testClaimValidator.shouldRefresh(tokenPayload) === false);
+        });
+    });
+
+    describe("Client Clock is ahead 1 hour of server time", () => {
+        before(function () {
+            DateProviderReference.init();
+        });
+
+        it("SessionClaimValidator.shouldRefresh should return true even after refreshing", function () {
+            const tokenPayload = {
+                "st-test": {
+                    v: false,
+                    t: Date.now()
+                }
+            };
+
+            const testClaim = new BooleanClaim({
+                id: "st-test",
+                refresh: () => {
+                    tokenPayload["st-test"].t = Date.now();
+                },
+                defaultMaxAgeInSeconds: 10 /* 10 seconds */
+            });
+
+            const testClaimValidator = testClaim.validators.isTrue();
+
+            withFakeClock(Date.now() + ONE_HOUR_IN_MS, () => {
+                assert(testClaimValidator.shouldRefresh(tokenPayload) === true);
+            });
+
+            testClaimValidator.refresh();
+
+            withFakeClock(Date.now() + ONE_HOUR_IN_MS, () => {
+                assert(testClaimValidator.shouldRefresh(tokenPayload) === true);
+            });
+        });
+
+        it("SessionClaimValidator.shouldRefresh should return false after refreshing", function () {
+            const tokenPayload = {
+                "st-test": {
+                    v: false,
+                    t: Date.now() - 20000 // setting claim fetched time to 20 seconds ago
+                }
+            };
+
+            const testClaim = new BooleanClaim({
+                id: "st-test",
+                refresh: () => {
+                    tokenPayload["st-test"].t = Date.now();
+                },
+                defaultMaxAgeInSeconds: 10 /* 10 seconds */
+            });
+
+            const testClaimValidator = testClaim.validators.isTrue();
+
+            // Adjust DateProvider clientClockDeviation
+            DateProviderReference.getReferenceOrThrow().dateProvider.setClientClockDeviationInMillis(-ONE_HOUR_IN_MS);
```

Comment:
You can move this into the `refresh` function to "simulate" this being done by response interception.

Reply:
done